service/history/timerSequence.go

```diff
@@ -113,6 +113,12 @@ func (t *timerSequenceImpl) createNextUserTimer() (bool, error) {
 
 	firstTimerTask := sequenceIDs[0]
 
+	// user timer after workflow timeout, skip
+	workflowRunExpirationTime := timestamp.TimeValue(t.mutableState.GetExecutionInfo().RetryExpirationTime)
```

Comment:
`RetryExpirationTime` is something which applies to multiple runs of workflow execution.  I think this check should be made against single run.